pkg/shipper/shipper.go

```diff
@@ -383,7 +389,25 @@ func (s *Shipper) upload(ctx context.Context, meta *metadata.Meta) error {
 	if err := meta.WriteToDir(s.logger, updir); err != nil {
 		return errors.Wrap(err, "write meta file")
 	}
-	return block.Upload(ctx, s.logger, s.bucket, updir, s.hashFunc)
+
+	err := block.Upload(ctx, s.logger, s.bucket, updir, s.hashFunc)
+
+	if err != nil {
+		return errors.Wrap(err, "while upploading the block")
+	}
+
+	files, err := block.GatherFileStats(updir, s.hashFunc, s.logger)
```

Comment:
Maybe `fileStats` would be a better name?

Reply:
The proposed name is easier to understand so I will change it. Thanks for the suggestion!